npm/npm.go

```diff
@@ -157,14 +157,22 @@ func (npMgr *NetworkPolicyManager) restore() {
 }
 
 // backup takes snapshots of iptables filter table and saves it periodically.
-func (npMgr *NetworkPolicyManager) backup() {
-	iptMgr := iptm.NewIptablesManager()
-	var err error
-	for {
-		time.Sleep(backupWaitTimeInSeconds * time.Second)
+func (npMgr *NetworkPolicyManager) backup(stopCh <-chan struct{}) {
+	var (
+		err    error
+		iptMgr = iptm.NewIptablesManager()
+		ticker = time.NewTicker(time.Second * time.Duration(backupWaitTimeInSeconds))
+	)
+	defer ticker.Stop()
```

Comment:
don't declare variables like this, it's preferred to use the one-line declare and initialize whenever you can (and you can, here)
```suggestion
	iptMgr := iptm.NewIptablesManager()
	ticker := time.NewTicker(time.Second * time.Duration(backupWaitTimeInSeconds))
	defer ticker.Stop()
```

and do the same thing for the error in the inner for-select scope, you don't need it at the function level